reactioncommerce/reaction
# Fix #4502: null check in email validation

Resolves #4502  
Impact: **minor**  
Type: **bugfix**

## Issue
When login/register was clicked without filling username and password the spinner runs forever.
`email.trim()` was being called without putting a null check on email and email was passed on as null.

## Solution
Added null check before calling `email.trim()`. This causes the validation to fail in case of empty email.

## Breaking changes
None.

## Testing
1. As an anonymous consumer, add a product to your cart.
2. Checkout.
3. Leaving email and password fields empty, click Register.
4. Observe the validation errors.
5. Repeat the same for Sign In
